ESP8266Interface.cpp

```diff
@@ -67,6 +67,30 @@ int ESP8266Interface::connect()
     return NSAPI_ERROR_OK;
 }
 
+nsapi_error_t ESP8266Interface::gethostbyname(const char* name, SocketAddress *address, nsapi_version_t version)
+{
+    if (address->set_ip_address(name)) {
+        if (version != NSAPI_UNSPEC && address->get_ip_version() != version) {
+            return NSAPI_ERROR_DNS_FAILURE;
+        }
+
+        return NSAPI_ERROR_OK;
+    }
+    
+    char* ipbuff = (char*)malloc(256);
```

Comment:
Is this a string representation of an IP address? We have a define for the max size you should use ([NSAPI_IP_SIZE](https://github.com/ARMmbed/mbed-os/blob/master/features/netsocket/nsapi_types.h#L88-L90)).

Also you should either check for out of memory (malloc returns NULL) or use `new char[blah]` (asserts on oom).